lib/plugins/aws/package/compile/functions/index.js

```diff
@@ -210,12 +210,27 @@ class AwsCompileFunctions {
         }
       }
 
+      let kmsKeyArn;
       const serviceObj = this.serverless.service.serviceObject;
-      if (functionObject.awsKmsKeyArn || (serviceObj && serviceObj.awsKmsKeyArn)) {
-        const arn = functionObject.awsKmsKeyArn || (serviceObj && serviceObj.awsKmsKeyArn);
+      if (functionObject.kmsKeyArn || this.serverless.service.provider.kmsKeyArn) {
+        kmsKeyArn = functionObject.kmsKeyArn || this.serverless.service.provider.kmsKeyArn;
+      } else if (functionObject.awsKmsKeyArn) {
+        this.serverless._logDeprecation(
+          'FUNCTION_DEPRECATED_PROPERTY_AWS_KMS_KEY_ARN',
+          'Starting with next major version, awsKmsKeyArn function property will be replaced by kmsKeyArn.'
+        );
+        kmsKeyArn = functionObject.awsKmsKeyArn;
+      } else if (serviceObj && serviceObj.awsKmsKeyArn) {
```

Comment:
`serviceObj` is guaranteed to be here (although it's possible it's written that way as tests work on incomplete instances).

Anyway point is, that if relying on `serviceObj.awsKmsKeyArn` unconditionally won't break the existing tests, let's not check for `serviceObj` existence

Reply:
57 tests do fail if serviceObj is not checked for existence.

Reply:
Ok, thanks for checking, let's leave it then (tests require refactor to `runServerless` but ofc that's not in scope here)